pyfolio/timeseries.py

```diff
@@ -491,95 +493,129 @@ def stability_of_timeseries(returns):
+       The starting returns (default 1).
 
     Returns
     -------
-    float
-        Similarity between returns.
+    pandas.Series
+        Series of cumulative returns.
 
+    Notes
+    -----
+    For increased numerical accuracy, convert input to log returns
+    where it is possible to sum instead of multiplying.
     """
 
-    bt_ts_pct = bt_ts.dropna()
-    oos_ts_pct = oos_ts.dropna()
+    # df_price.pct_change() adds a nan in first position, we can use
+    # that to have cum_returns start at the origin so that
+    # df_cum.iloc[0] == starting_value
+    # Note that we can't add that ourselves as we don't know which dt
+    # to use.
+    if pd.isnull(returns.iloc[0]):
+        returns.iloc[0] = 0.
 
-    bt_ts_r = bt_ts_pct.reshape(len(bt_ts_pct), 1)
-    oos_ts_r = oos_ts_pct.reshape(len(oos_ts_pct), 1)
+    df_cum = np.exp(np.log(1 + returns).cumsum())
 
-    if transform_style == 'raw':
-        bt_scaled = bt_ts_r
-        oos_scaled = oos_ts_r
-    if transform_style == 'scale':
-        bt_scaled = preprocessing.scale(bt_ts_r, axis=0)
-        oos_scaled = preprocessing.scale(oos_ts_r, axis=0)
-    if transform_style == 'normalize_L2':
-        bt_scaled = preprocessing.normalize(bt_ts_r, axis=1)
-        oos_scaled = preprocessing.normalize(oos_ts_r, axis=1)
-    if transform_style == 'normalize_L1':
-        bt_scaled = preprocessing.normalize(bt_ts_r, axis=1, norm='l1')
-        oos_scaled = preprocessing.normalize(oos_ts_r, axis=1, norm='l1')
+    if starting_value is None:
+        return df_cum - 1
+    else:
+        return df_cum * starting_value
 
-    X_train = bt_scaled
-    X_test = oos_scaled
 
-    X_train = X_train.reshape(len(X_train))
-    X_test = X_test.reshape(len(X_test))
+def aggregate_returns(df_daily_rets, convert_to):
+    """
+    Aggregates returns by week, month, or year.
 
-    x_axis_dim = np.linspace(-4, 4, 100)
-    kernal_method = 'scott'
+    Parameters
+    ----------
+    df_daily_rets : pd.Series
+       Daily returns of the strategy, noncumulative.
+        - See full explanation in tears.create_full_tear_sheet (returns).
+    convert_to : str
+        Can be 'weekly', 'monthly', or 'yearly'.
 
-    try:
-        scipy_kde_train = stats.gaussian_kde(
-            X_train,
-            bw_method=kernal_method)(x_axis_dim)
-        scipy_kde_test = stats.gaussian_kde(
-            X_test,
-            bw_method=kernal_method)(x_axis_dim)
-    except:
-        if return_zero_if_exception:
-            return 0.0
-        else:
-            return np.nan
+    Returns
+    -------
+    pd.Series
+        Aggregated returns.
+    """
 
-    kde_diff = sum(abs(scipy_kde_test - scipy_kde_train)) / \
-        (sum(scipy_kde_train) + sum(scipy_kde_test))
+    def cumulate_returns(x):
+        return cum_returns(x)[-1]
 
-    return kde_diff
+    if convert_to == WEEKLY:
+        return df_daily_rets.groupby(
+            [lambda x: x.year,
+             lambda x: x.month,
```

Comment:
I don't think we need the month, the isocalendar week field is the weeknumber in the year,